python/ray/experimental/channel/common.py

```diff
@@ -41,29 +43,22 @@ def create_channel(
         """
         raise NotImplementedError
 
+    def requires_nccl(self) -> bool:
+        # By default, channels do not require NCCL.
+        return False
 
-def _do_register_custom_serializers(
-    self: Any, channel_output_types: List[type]
-) -> None:
-    """
-    Register custom serializers for the given channel types. This method should
-    be run on the reader(s) and writer of a channel, which are the driver
-    and/or Ray actors.
-
-    Args:
-        self: This method should be run on the driver or the Ray actor. The Ray
-            actor should be passed as `self`.
-        channel_output_types: The list of channel output types to register.
-    """
-    for typ in channel_output_types:
-        typ.register_custom_serializer(self)
+    def set_nccl_group_id(self, group_id: str) -> None:
+        raise NotImplementedError
 
 
 @DeveloperAPI
 @dataclass
 class ChannelContext:
-    # Used for the torch.Tensor NCCL transport.
-    nccl_group: Optional["_NcclGroup"] = None
+    serialization_context = _SerializationContext()
```

Comment:
Why do we need to keep it here btw? Isn't it cleaner to just create it on the fly whenever serialization is needed?

Reply:
Just for convenience. The ChannelContext will get created on the fly.